SmartCart.DataProvider/Repositories/CartRepository/CartRepository.cs

```diff
@@ -0,0 +1,119 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using SmartCart.DataProvider.DatabaseContext;
+using SmartCart.DataProvider.Models;
+
+namespace SmartCart.DataProvider.Repositories
+{
+    public class CartRepository : ICartRepository
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CartRepository(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<bool> AddAsync(CartDto cart)
+        {
+            if(cart == null)
+            {
+                return false;
+            }
+
+            var cartEntity = _mapper.Map<Cart>(cart);
+
+            await _context.Carts.AddAsync(cartEntity);
+            await _context.CartMembers.AddAsync(new CartMember
+            {
+                CartID = cartEntity.CartID,
+                MemberID = cartEntity.CartOwnerID
+            });
+
+            var result = await _context.SaveChangesAsync();
+
+            return result > 0;
+        }
+
+        public async Task<bool> AddAsync(List<CartDto> carts)
```

Comment:
Raise an exception, as program now in suspicious and unpredictable state. What does false means? Database cannot add or objects are corrupted?